fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -331,14 +331,28 @@ private Statistics estimateEqualTo(ComparisonPredicate cp, ColumnStatistic stats
             ColumnStatistic statsForRight,
             EstimationContext context) {
         double selectivity;
-        double ndv = statsForLeft.ndv;
-        double val = statsForRight.maxValue;
-        if (val > statsForLeft.maxValue || val < statsForLeft.minValue) {
-            selectivity = 0.0;
+        if (statsForLeft.isUnKnown) {
+            selectivity = DEFAULT_INEQUALITY_COEFFICIENT;
         } else {
-            selectivity = StatsMathUtil.minNonNaN(1.0, 1.0 / ndv);
+            double ndv = statsForLeft.ndv;
+            if (statsForRight.isUnKnown) {
+                if (ndv >= 1.0) {
```

Comment:
what is the case where statsForLeft is not unknown but ndv < 1.0? Should we do a normalization if the ndv < 1.0?

Reply:
this ndv is from left, and right is unknown

Reply:
I mean whether we should do a data normalization for those concept which has certain physical properties, such as row count, ndv, selectivity, etc. We may do this at next generation code model.